### PR TITLE
chore: upgrade to @zendeskgarden/scripts v2

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Upgrade to ESM.

## Detail

https://github.com/zendeskgarden/scripts/pull/180

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "@zendeskgarden/react-grid": "8.69.3",
     "@zendeskgarden/react-theming": "8.69.3",
     "@zendeskgarden/react-typography": "8.69.3",
-    "@zendeskgarden/scripts": "1.4.3",
+    "@zendeskgarden/scripts": "2.0.1",
```
Comment (geotrev) [inline package.json:46]: Does this file need `"type": "module"` similar to [scripts.git](https://github.com/zendeskgarden/scripts/blob/main/package.json#L12)?

File: .circleci/config.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
     steps:
       - *attach_workspace
       - run: yarn build:demo
-      - run: scripts/deploy.js
+      - run: scripts/deploy.mjs
```
Comment (geotrev) [inline .circleci/config.yml:47]: Bummer the `mjs` extension is needed - maybe specifying `type` in `package.json` makes it work with `js`? 🤞🏻

File: package.json (hunk 1 above)
Comment (jzempel) [reply package.json:46]: I don't think we want to release `@zendeskgarden/svg-icons` as an ESM module. The goal of the current task is to incorporate the ESM module `@zendeskgarden/scripts` into each of the repos. Generally speaking, the upgrades will feel more like https://www.gatsbyjs.com/docs/how-to/custom-configuration/es-modules/.

File: .circleci/config.yml (hunk 2 above)
Comment (jzempel) [reply .circleci/config.yml:47]: Important to note that the package `main` is `dist/index.svg`. We're not packaging an ESM module here. Again, referencing https://www.gatsbyjs.com/docs/how-to/custom-configuration/es-modules/, I _think_ using `.mjs` is the desired approach.

Comment (geotrev) [reply .circleci/config.yml:47]: Good call out. Thx for clarifying.

**Review by geotrev — APPROVED** [review]

**Pull request merged**